Pinging elasticsearch before index

ahmetcetin opened the pull request:
Pinging elasticsearch before index not to crash

ahmetcetin committed:
pinging elasticsearch before indexing

ahmetcetin committed:
pinging elasticsearch before indexing

ahmetcetin committed:
pinging elasticsearch before indexing

moooji commented:
Thx for the PR!

The issue that I see is that this PR will swallow potential write errors :-o and simply console log them. IMHO errors should be thrown/returned so that the application can handle them properly.

Ideally, we would like to retry the write when the cluster was temporarily down.

Also, it seems unnecessary to ping before write, since if the cluster is down, write will return an error exactly like ping does. I cannot see the benefit of the added ping, but maybe I am missing something here. 

Can you maybe explain your scenario / use case a bit more?

gfogle commented:
this could be potentially useful for say container-based environments where i may want to do a health check prior to marking something g2g.

for instance if im doing container rotations i want to make sure that something that was configured isnt broken after ive taken prior, healthy containers down.

ahmetcetin committed:
es.ping method change in es5

sending hello parameter in ping method causes es to return 400 in version 5, so removed.
